packages/lesswrong/components/tagging/TagsTooltip.tsx

```diff
@@ -54,74 +105,81 @@ const DefaultPreviewWrapper: TagsTooltipPreviewWrapper = ({children}) => (
-  const { results: pingbacks, loading: loadingPingbacks } = useRedLinkPingbacks(tag?._id);
-  const title = inferRedLinkTitle(tag, slug??null);
-
-  return <div className={classes.redLinkTooltip}>
-    <Typography variant='title'>
-      {title}
-    </Typography>
-    <ContentStyles contentType='tag'>
-      This red link was used on {pingbacks ? pingbacks.length : <Loading/>} page{pingbacks ? (pingbacks.length > 1 ? "s" : "") : "(s)"}:
-      <ul>
-        {pingbacks?.map(pingback => <li key={pingback._id}>
-          <TagHoverPreview
-            targetLocation={{ params: { slug: pingback.slug }, hash: '', query: {} } as AnyBecauseTodo}
-            href={tagGetUrl({slug: pingback.slug})} noPrefetch
-          >
-            <Link to={tagGetUrl({slug: pingback.slug})}>
-              {pingback.name}
-            </Link>
-          </TagHoverPreview>
-        </li>)}
-      </ul>
-    </ContentStyles>
-    <ContentStyles contentType='tag' className={classes.redLinkTooltipTitle}>
-      A red link highlights author's intention to point at a concept that doesn't have a satisfactory explanation written (by their standards).
-    </ContentStyles>
-  </div>
-}
+  const { tag: currentlyViewingTag } = useTagPageContext() ?? {};
+  const excludedPingbackTagIds = currentlyViewingTag?._id ? [currentlyViewingTag._id] : undefined;
+  const { results: pingbacks, loading: loadingPingbacks } = useRedLinkPingbacks(tag?._id, excludedPingbackTagIds);
+  const title = inferRedLinkTitle(tag, slug ?? null);
+
+  if (!tag) {
+    return (
+      <div className={classes.noTagOrPlaceholderMessage}>
+        <ContentStyles contentType='tag'>
+          No page or placeholder found for this link.
+        </ContentStyles>
+      </div>
+    );
+  }
+
+  let pingbacksSection;
+
+  if (loadingPingbacks) {
+    // Display loading indicator while pingbacks are loading
+    pingbacksSection = <Loading />;
+  } else if (pingbacks && pingbacks.length > 0) {
+    // Handle singular and plural forms for pingbacks count
+    const pingbackCount = pingbacks.length;
+    const pageWord = pingbackCount === 1 ? 'page' : 'pages';
+
+    pingbacksSection = (
+      <div>
+        This red link was used on {pingbackCount} other {pageWord}:
+        <ul>
+          {pingbacks.map(pingback => (
+            <li key={pingback._id}>
+              <TagHoverPreview
+                targetLocation={{ params: { slug: pingback.slug }, hash: '', query: {} } as AnyBecauseTodo}
+                href={tagGetUrl({ slug: pingback.slug })}
+                noPrefetch
+              >
+                <Link to={tagGetUrl({ slug: pingback.slug })}>
+                  {pingback.name}
+                </Link>
+              </TagHoverPreview>
+            </li>
+          ))}
+        </ul>
+      </div>
+    );
+  } else {
+    // Display message when there are no pingbacks
+    pingbacksSection = (
+      <div>
+        This is a red link.
+      </div>
+    );
+  }
```

Comment:
A few notes:
- the third (last) block should be second, for ease of readability
- this entire thing should be a component rather than a conditional assignment, given how complicated it is